docs/scripts/generate-plugins.py

```diff
@@ -143,30 +143,31 @@ def main() -> None:
     tmt.plugins.explore(logger)
 
     if step_name == 'discover':
-        plugin_generator = _create_step_plugin_iterator(
-            tmt.steps.discover.DiscoverPlugin._supported_methods)
+        parent_class = tmt.steps.discover.DiscoverPlugin
+        plugin_generator = _create_step_plugin_iterator(parent_class._supported_methods)
```

Comment:
Now that `parent_class` carries the difference between branches, we can drop `plugin_generator` creation from every branch and create it just once after `if` is done, before line 176.

Reply:
Good idea, changed in 9de48aa2.